### PR TITLE
new config for attribute "duration" ( MagicWall - Wild Growth duration random )

### PR DESCRIPTION
**Pull request by MillhioreBT**
With these changes, it is now possible to configure a range of durations for temporary articles, thus complying with what is mentioned in the PR #3148

> - [x] magic wall should have a random duration between 14 and 20 seconds when shot
> - [x]  wild growth should have a random duration between 38 and 45 seconds when shot

### CODE REVIEW

**Comment by EPuncker** [comment]
maybe add a warning if the first value is bigger than the second one

**Comment by MillhioreBT** [comment]
> maybe add a warning if the first value is bigger than the second one

`random_number` function reverses the order from least to greatest always, so it's not a problem, the warning would only be something for messy people, and in the base branch we will never include messy things, or at least that's what I think, if what I say not true let me know, i am prone to errors :(

File: data/items/items.xml [hunk 1]
```diff
@@ -686,15 +686,15 @@
 	<item id="1497" article="a" name="magic wall">
 		<attribute key="type" value="magicfield" />
 		<attribute key="decayTo" value="0" />
-		<attribute key="duration" value="20" />
+		<attribute key="duration" value="14;20" />
```
[the remark this answers is not in the document]
Comment (MillhioreBT) [reply data/items/items.xml:689]: <3

File: src/items.cpp [hunk 2]
```diff
@@ -819,8 +821,29 @@ void Items::parseItemNode(const pugi::xml_node& itemNode, uint16_t id)
...
+					auto split = explodeString(valueAttribute.as_string(), ";");
+					if (split.size() == 2) {
+						it.decayTime = pugi::cast<uint32_t>(split[0].c_str());
+						uint32_t value = pugi::cast<uint32_t>(split[1].c_str());
+						if (it.decayTimeMax) {
+							std::cout << "[Warning - Items::parseItemNode] decayTimeMax is being overwritten to: " << value << std::endl;
+						}
+						it.decayTimeMax = value;
+					}
+					else {
```
Comment (nekiro) [inline src/items.cpp:835]: this whole code is not needed as we are not using ; serparator for min and max
just
it.decayTime = pugi::cast<uint32_t>(valueAttribute.value());

**Review by nekiro — CHANGES_REQUESTED** [review]

File: src/items.cpp (hunk 2 above)
Comment (MillhioreBT) [reply src/items.cpp:835]: I think you hate the separator ahahahaha, but since you're the only one paying attention to me, I'm going to listen to you <3

**Review by lucasromagnoli — DISMISSED** [review]

**Comment by nekiro** [comment]
Hey @MillhioreBT care to resolve requested changes?

File: src/item.h [hunk 3]
```diff
@@ -967,6 +967,9 @@ class Item : virtual public Thing
 			}
 		}
 		uint32_t getDefaultDuration() const {
+			if (items[id].decayTimeMax) {
```
Comment (nekiro) [inline src/item.h:970]: This method gonna return random values now, how we gonna get durationmin and durationmax in this case? I think getDefaultDuration shouldn't be changed and getDefaultDurationMax should be added.

**Review by nekiro — CHANGES_REQUESTED** [review]
Also I think you missed luascript changes, we have no possibility to getDurationMax from itemtype now.
Edit: ah we didn't have them before too, but wont hurt to add these.

**Comment by MillhioreBT** [comment]
I have a problem and inadvertently i delete the repositories that had the changes, now that I want to complete the suggestions, it does not let me update this PR, I suppose I will have to open a totally new one

**Comment by Znote** [comment]
> I have a problem and inadvertently i delete the repositories that had the changes, now that I want to complete the suggestions, it does not let me update this PR, I suppose I will have to open a totally new one

If you need to recover this, I think you can pull/checkout otland/0178a3c7a3678c4b6b594db651c23bdd1943b7fc

**Pull request closed without merging**